decidim-blogs/spec/forms/decidim/blogs/admin/post_form_spec.rb

```diff
@@ -52,15 +54,35 @@ module Admin
             subject.map_model(post)
           end
 
-          it "does not assign the user group for normal users" do
-            expect(subject.user_group_id).to be_nil
+          it "assigns current_user as user_group_id" do
+            expect(subject.user_group_id).to eq("current_user")
+          end
+
+          it "assigns user name as author" do
+            expect(subject.author).to be_nil
+          end
+
+          context "when author is an organization" do
+            let(:author) { current_organization }
+
+            it "assigns 'current_organization' as user_group_id" do
+              expect(subject.user_group_id).to eq("current_organization")
+            end
+
+            it "assigns current_organization as author" do
+              expect(subject.author).to eq(author)
+            end
           end
 
           context "when the author is a group" do
             let(:author) { create(:user_group, :verified, organization: current_organization) }
 
-            it "assigns the user group ID to the form" do
-              expect(subject.user_group_id).to eq(author.id)
+            it "assigns user_group.id as user_group_id" do
+              expect(subject.user_group_id).to eq(author.id.to_s) # to_s to revise
+            end
+
+            it "assigns user_group.name as author" do
+              expect(subject.author).to eq(author)
```

Comment:
```suggestion
            it "assigns user_group.id as user_group_id" do
              expect(subject.user_group_id).to eq(user_group.id.to_s)
            end

            it "assigns user_group.name as author" do
              expect(subject.author).to eq(user_group)
```
Finally, here we do expect an `id` (as a string indeed), but the `author` method (that always return an object), should correspond to the `user_group` in this context.

Reply:
Done